connector/oauth/oauth.go

```diff
@@ -209,12 +209,12 @@ func (c *oauthConnector) HandleCallback(s connector.Scopes, r *http.Request) (id
 		return identity, fmt.Errorf("OAuth Connector: failed to parse userinfo: %v", err)
 	}
 
-	userID, found := userInfoResult[c.userIDKey].(string)
+	userID, found := userInfoResult[c.userIDKey]
 	if !found {
 		return identity, fmt.Errorf("OAuth Connector: not found %v claim", c.userIDKey)
 	}
 
-	identity.UserID = userID
+	identity.UserID = fmt.Sprintf("%v", userID)
```

Comment:
It seems ok for id to be a number or string, yet this piece of the code will also convert arrays and maps to strings.

I suggest using a custom logic to convert claims to strings.

```go
switch userID.(type) {
case float64, int64, string:
    identity.UserID = fmt.Sprintf("%v", userID)
default:
     return identity, fmt.Errorf("OAuth Connector: %v claim should be string or number, got %T", c.userIDKey, userID)
}
```

And one more thing. Maybe we can also apply the same logic to username and preferred username claims. It is ok to have a username like, e.g., 644587, but it is an open question.

Reply:
Done.